docs/adding-new-resource.md

```diff
@@ -0,0 +1,705 @@
+    logger.go:42: 14:32:49 | case/0-apply |       reason: ReconcileError
+    logger.go:42: 14:32:49 | case/0-apply |       status: "False"
+    logger.go:42: 14:32:49 | case/0-apply |       type: Synced
+```
+
+Make the fixes, create a [new commit], and trigger the automated test again.
+
+**Ignoring Some Resources in Automated Tests**
+
+Some resources require manual intervention such as providing valid public keys
+or using on-the-fly values. These cases can be handled in manual tests, but in
+cases where we cannot provide generic values for automated tests, we can skip
+some resources in the tests of the relevant group via an annotation:
+
+```yaml
+upjet.upbound.io/manual-intervention: "The Certificate needs to be provisioned successfully which requires a real domain."
+```
+
+The key is important for skipping, we are checking this
+`upjet.upbound.io/manual-intervention` annotation key and if is in there, we
+skip the related resource. The value is also important to see why we skip this
+resource.
+
+```
+NOTE: For resources that are ignored during Automated Tests, manual testing is a
+must. Because we need to make sure that all resources published in the `v1beta1`
+version is working.
+```
+
+## Manual Test
+
+Configured resources can be tested by using manual methods. This method generally
+contains the environment preparation and creating the example manifest in the
+Kubernetes cluster steps. The following steps can be followed for preparing the
+environment:
+
+1. Registering the CRDs (Custom Resource Definitions) to Cluster: We need to
+apply the CRD manifests to the cluster. The relevant manifests are located in
+the `package/crds` folder of provider subdirectories such as:
+`provider-aws/package/crds`. For registering them please run the following
+command: `kubectl apply -f package/crds`
+
+2. Create ProviderConfig: ProviderConfig Custom Resource contains some
+configurations and credentials for the provider. For example, to connect to the
+cloud provider, we use the credentials field of ProviderConfig. For creating the
+ProviderConfig with correct credentials, please see:
+
+- [Create a Kubernetes secret with the AWS credentials]
+- [Create a Kubernetes secret with the Azure credentials]
+- [Create a Kubernetes secret with the GCP credentials]
+
+3. Start Provider: For every Custom Resource, there is a controller and these
+controllers are part of the provider. So, for starting the reconciliations for
+Custom Resources, we need to run the provider (collect of controllers). For
+running provider: Run `make run`
+
+4. Now, you can create the examples you've generated and check events/logs to
+spot problems and fix them.
+
+- Start Testing: After completing the steps above, your environment is ready for
+testing. There are 3 steps we need to verify in manual tests: `Apply`, `Import`,
+`Delete`.
+
+**Apply:** We need to apply the example manifest to the cluster.
```

Comment:
```suggestion
### Apply

We need to apply the example manifest to the cluster.
```

Question: Shall we style these “Apply”, “Import”, “Delete” sections as level 3 headers? If so, we can apply the same styling to “Import” and “Delete” titles below.